packages/core/src/odata-v4/entity-serializer.spec.ts

```diff
@@ -117,4 +118,39 @@ describe('entity-serializer', () => {
       TimeOfDayProperty: '01:02:03.456'
     });
   });
+
+  it('should include custom fields on navigation properties', () => {
+    const testEntity = TestEntity.builder()
+      .toSingleLink(
+        TestEntitySingleLink.builder()
+          .withCustomFields({ custom: 'custom' })
+          .build()
+      )
+      .build();
+    expect(serializeEntityV4(testEntity, TestEntity)).toEqual({
+      to_SingleLink: {
+        custom: 'custom'
+      }
+    });
+  });
+
+  it('should consider only changed items when diff is true', () => {
```

Comment:
These are not v4 specific, right? Could we move them to v2/common then?

Reply:
Also created follow up for this, as I think this is a bigger task, if we want to do it right ;)